src/display/annotation_layer.js

```diff
@@ -1832,157 +1847,234 @@ class ChoiceWidgetAnnotationElement extends WidgetAnnotationElement {
+  #dateTimePromise = null;
 
-    this.pinned = false;
-  }
+  #boundHide = this.#hide.bind(this);
 
-  render() {
-    const BACKGROUND_ENLIGHT = 0.7;
+  #boundShow = this.#show.bind(this);
+
+  #boundToggle = this.#toggle.bind(this);
+
+  #color = null;
+
+  #container = null;
+
+  #contentsObj = null;
+
+  #elements = null;
+
+  #parent = null;
+
+  #parentRect = null;
+
+  #pinned = false;
+
+  #popup = null;
+
+  #rect = null;
+
+  #richText = null;
+
+  #titleObj = null;
+
+  constructor({
+    container,
+    color,
+    elements,
+    titleObj,
+    modificationDate,
+    contentsObj,
+    richText,
+    parent,
+    rect,
+    parentRect,
+    open,
+  }) {
+    this.#container = container;
+    this.#titleObj = titleObj;
+    this.#contentsObj = contentsObj;
+    this.#richText = richText;
+    this.#parent = parent;
+    this.#color = color;
+    this.#rect = rect;
+    this.#parentRect = parentRect;
+    this.#elements = elements;
+
+    const dateObject = PDFDateString.toDateObject(modificationDate);
+    if (dateObject) {
+      // The modification date is shown in the popup instead of the creation
+      // date if it is available and can be parsed correctly, which is
+      // consistent with other viewers such as Adobe Acrobat.
+      this.#dateTimePromise = parent.l10n.get("annotation_date_string", {
+        date: dateObject.toLocaleDateString(),
+        time: dateObject.toLocaleTimeString(),
+      });
+    }
 
-    const wrapper = document.createElement("div");
-    wrapper.classList.add("popupWrapper");
+    this.trigger = elements.flatMap(e => e.getElementsToTriggerPopup());
+    // Attach the event listeners to the trigger element.
+    for (const element of this.trigger) {
+      element.addEventListener("click", this.#boundToggle);
+      element.addEventListener("mouseenter", this.#boundShow);
+      element.addEventListener("mouseleave", this.#boundHide);
+      if (typeof PDFJSDev !== "undefined" && PDFJSDev.test("TESTING")) {
+        element.classList.add("popupTriggerArea");
+      }
+    }
 
-    // For Popup annotations we hide the entire section because it contains
-    // only the popup. However, for Text annotations without a separate Popup
-    // annotation, we cannot hide the entire container as the image would
-    // disappear too. In that special case, hiding the wrapper suffices.
-    this.hideElement = this.hideWrapper ? wrapper : this.container;
-    this.hideElement.hidden = true;
+    this.#container.hidden = true;
+    if (open) {
+      this.#toggle();
+    }
 
-    const popup = document.createElement("div");
-    popup.classList.add("popup");
+    if (typeof PDFJSDev === "undefined" || PDFJSDev.test("TESTING")) {
+      // Since the popup is lazily created, we need to ensure that it'll be
+      // created and displayed during reference tests.
+      this.#parent.popupShow.push(() => {
+        if (this.#container.hidden) {
+          this.#show();
+        }
```

Comment:
Given that you've re-factored how the l10n is handled for Popups, is there not potentially a small risk that the ref-test could fail intermittently because we no longer wait for Popup-l10n to complete during tests?

How about something like the following to avoid that (further comments below)?
```suggestion
      this.#parent.popupShow.push(async () => {
        if (this.#container.hidden) {
          this.#show();
        }
        await #dateTimePromise;
```